### ISSUE TITLE
I know it's crazy, maybe `UnoCSS` support? 

### ISSUE THREAD

**Issue by zernonia** [radix-vue/shadcn-vue#34]
🙈🤪

**Comment by JessicaSachs** [comment]
You can extend an UnoCSS preset to solve this issue. I would start with the Tailwind one. It's actually fairly easy! Just a lot of regexp.

I'm not sure if someone wants to update the @unocss/preset-tailwind package but that's where I'd start.

It's a little hard for me because I don't have a ton of edge-case Tailwind understanding. The basics, I got. But some of the weirder syntax is not easy for me to list out.

**Comment by zernonia** [comment]
Thanks for the input @JessicaSachs ! I believe `shadcn-vue` would be a good reference point for testing out edge-case which was not included in the **preset** package, since there's quite many components, and some with weird class 😁

Other than the syntax, there's also some tweak needed to be done when initiating the project. Instead of `tailwind.config.js` and tw plugins, it should import `unocss.config.ts`.

**Comment by JessicaSachs** [comment]
Uno automatically looks for a tailwind config btw and iirc they merge them

**Comment by miguelrk** [comment]
**FYI:** the [fisand/unocss-preset-shadcn](https://github.com/fisand/unocss-preset-shadcn) preset might be a good starting point. Though I believe its also missing advanced Tailwindcss features

**Comment by Dunqing** [comment]
I'm interested in this, but maybe I need more time to do this.

**Comment by JessicaSachs** [comment]
Will be watching eagerly!!! Huge win for Vue ecosystem if we can get compatibility. A lot of Vue uses Uno because of @antfu influence

**Comment by JessicaSachs** [comment]
I wrote up some steps in my Discord for someone looking for something to work on, but I'm not sure if he's got interest or bandwidth to work on it. I figured I'd post it publicly.

### Actual
The examples inside of the radix-vue documentation don't work because of lack of support for a lot of the more complex TailwindCSS `/` for opacity, `group-hover-data` or as UnoCSS would do `group:hover:data-`.

1. There's examples [in the Accordion](https://www.radix-vue.com/components/accordion.html#accordion) that I want to work with `unocss.config.js` files, not just `tailwind.config.js` files, since UnoCSS is basically a more extendable Tailwind compiler.
2. In the [radix-vue](https://www.radix-vue.com/overview/getting-started.html) docs, they assume you're using Tailwind and give you the correct tailwind config to make it happen.

I'd love it if Uno supported this in their main preset.

### Issue

1. When you plug these examples into UnoCSS, they don't style correctly. UnoCSS's [default preset attempts to be a superset of Tailwind](https://unocss.dev/presets/uno#usage), so it seems like fixing these issues is in-scope.
    1. The default preset attempts to be compatible with Tailwind always, but it seems they're a little behind!
    1. Whoever picks this up should look at the more complex rules inside of the [radix-vue](https://www.radix-vue.com/components/accordion.html#accordion) link I linked above and make sure that all of the more advanced syntax used is supported.

### Solution
This can be accomplished by **extending the UnoCSS variants using Regexp**.

### Development Approach
Whoever picks this up can iterate quickly in two ways:
1. Without downloading any packages by writing code inline in the Playground
    1. [Playground link](https://uno.antfu.me/play/?html=DwEwlgbgBAFgtAMwK4BsVQC4FMAeG4DGWAdtgE5QIq5QDOW1B+xA9sVlAIZoBcGZnYrTAYwbOABYADFIB8AKChRQkKAFtOCpUpXRseOAFYc6BAHcAjDK7EwG7HABGLJMSJxuGG3c4OCL0jgwYgRgkQ5BHwcQJAFRcQtaLW0lAFVWAGEAZSzFbWAAenAIZJ1iqBYABwBmKUxcfBQAc0ozWrq1C1LtABUYDmDaDEEvcRAsDWIQKABBDBY1MAIobKyoEibgrAA6PJ0iyG7ddQAmSmocKAArJCGwBABPQhJyeoMTkwqaupgWCCwyABeABEVQAHFJgd0dJw9ikoGBCJwyM5iHAUCwmiw4JsMDAkI44Sl9Phgv0yCIidoYGQsAgQTAMBhKrQeAUCrj8Y5tv41AVXCwCLRaPzWELaMCqUphmQmlgMCCAPqOFCCADWkvhUFkhU0cMKxWSBsO8mNJVN5U4jloLBQSGwUGcTIWRigFKajLgdWoCHwdRJz1IAK+tVa7WSYwmgmmAHaEUJhqQoHHkGgHlACLd5osAF5W6img4lIA&config=JYWwDg9gTgLgBAbwFBzgEwKYDNgDsMDCEuOA5gDQpxhQYDOGMAgjDFMAEYCuMwWAnpVQ16jAJIBjYnSHVaDGAFVcESgF84WKBBBwA5FxUS6dPUiRTcdeCH4AFeYzgBeOAAoAlC4B8iKgHp-OABNCC4oOClMOAALDFoAQipaGHDcP1RUXABDEAwALn0MAA8YDFxMNABaGGzgABsAdzw0PVlUADds9mzcGDpCgG0qTLhAuAAhbLpgCWz6+v44fjC9WkjiDvjeXFI4WlISsB9IkxHM8YBRYtyweoLY1jABwMMIYzoAOkwO-0syfxdHp9OgAYhKt3u5lGqHGAAlupgIhDwPd3CsuHoFnBGr14DAIJF6sQMHAYHF9hhIJpoGSKTwGnQPPlzrCgnYuNiAEoYACOXHoMEKMSeL38pGA5K4HE+UhA-jeHwVRhM-jAnPq-gATAAOAAsAE5-Dh7mC0HwsFUtQA2AAMGA4HGyOq1aGyAGY0Hq9baDRwtRIsFg9e6sGg0AB2bIARiwBuw0YArNGPVqMBGdQbXe6JAboxhPRwJLafXM9Vho9auVqtVUue7recALpUNRINtII7QeCYLDZTk97B4QjEMhuZCoKCc+hDc6DPQSLjWHRVKf3NqIDbEqCFNYYVpwNRN2THqh0GLdxf9QoTzILpcEkBVc+Xnh6XdlUpVep7T8wKrQL0hyxBAWw7n+NQYPMZioGosgiAoAxwMMoy2A4ogwJ47RyBhygQFh5wIYwLBsJwDICARoxETAkjSOOrJwHQcz3IU0afFq2GoBIaC4LuIowM8+SBPQICfOe-gwZkageCe7YeEgQA&css=PQKgBA6gTglgLgUzAYwK4Gc4HsC2YDCAyoWABYJQIA0YAhgHYAmYcUD6AZllDhWOqgAOg7nAB0YAGLcwCAB60cggDYIAXGBDAAUKDBi0mXGADe2sGC704AWgDuCGAHNScDQFYADJ4Dc5sAACtMLKAJ5gggCMLPK2ABR2pPBIcsoAlH4WAEa0yADWTlBYqEw2yFjK3Bpw5LxxAOTllVDoYpSMYgAs3vUZ2gC+msBAA&options=N4XyA)
    1. Here's [how to extend variants](https://unocss.dev/config/variants#example). This is how they supported `hover:bg-red-500`.
      1. For an example: here is the PR in which they began supporting `group-data-` variant.
    1. However UnoCSS's preset is [missing some more variants](https://github.com/unocss/unocss/pull/2849/files#diff-260ebba82da3d4409b2cff43fdd7a1f9ef151a32e7892d3c91e3dbc040ca4f16R22-R36) that are present inside of the radix-vue package. For example, the `hover:group:data-` variant is missing.

### Scope + Benefit
This will help all users of UnoCSS, which is not limited to Vue users!

### Issue Tracking
Arguable, this issue could be moved into the UnoCSS repository.

**Comment by zouhangwithsweet** [comment]
Yeah. I am working on `unocss-preset-shadcn` to support vue-shadcn. Just test `button` & `accordion`. It works well.
Check: https://github.com/fisand/unocss-preset-shadcn/blob/main/vue-shadcn/App.vue. And I will test all components in the near future.

**Comment by a982246809** [comment]
现在我已有的项目已经使用了unocss , 再去安装tailwindcss 是不是冗余了?
Now that I have an existing project that uses unocss, is it redundant to install tailwindcss?

**Comment by zouhangwithsweet** [comment]
> 现在我已有的项目已经使用了unocss , 再去安装tailwindcss 是不是冗余了? Now that I have an existing project that uses unocss, is it redundant to install tailwindcss?

Yeah. Actually `unocss-preset-shadcn` has supported `vue-shadcn` but not perfect. Just need to add some special variants. Just check [quick-start](https://github.com/fisand/unocss-preset-shadcn#quick-start)

**Comment by orenmizr** [comment]
unocss-preset-shadcn works normal  with nuxt + unocss module + shadcn-vue ? is it stable ?

**Comment by orenmizr** [comment]
i removed nuxt+tailwind for nuxt+uno with shadcn. i noticed i did not get the reset css or fonts and probably more after installing all the deps. 

when adding shadcn to the project it creates a tailwind config that is probably responsible for the missing parts. 
how do i connect the uno so i don't need the tailwind config file ? 
 
![image](https://github.com/radix-vue/shadcn-vue/assets/642564/96e47f13-e1e3-4fdd-9f1b-c48820ac307f)

**Comment by sadeghbarati** [comment]
@orenmizr  Install `@unocss/reset` and then import it to your main js entry or a css file

https://github.com/unocss/unocss/blob/main/packages/reset/tailwind.md

**Comment by orenmizr** [comment]
i did. imported first via js, then via style. i saw it loaded in devtools. but the body margin was still there, 

i have another project with similar setup  but with tailwind+nuxt and there it looks polished- both have the  same tailwind code. wondering about the difference. like where the default  fonts come from... 

update: the tailwind reset fixed it. @import '@unocss/reset/tailwind.css';

**Comment by Dunqing** [comment]
If possible I think we could improve the `cli` implementation to support selecting `unocss/tailwindcss` and do the work of generating the config file. This could simplify the cost of using `unocss`

**Comment by zouhangwithsweet** [comment]
> unocss-preset-shadcn works normal with nuxt + unocss module + shadcn-vue ? is it stable ?

Absolute yes. I had check it. Just submit issue when you have it.

**Comment by orenmizr** [comment]
i am already using it :+1:  
btw, once init-ed ,shadch-vue  creates a component.json that points to tailwind.config.ts... not sure what to do with it

**Comment by zouhangwithsweet** [comment]
> If possible I think we could improve the `cli` implementation to support selecting `unocss/tailwindcss` and do the work of generating the config file. This could simplify the cost of using `unocss`

I have thought about this before. I think css engine should be optional in `cli`. But `unocss` still don’t have enough users compared with `tailwind`. But for a `unocss` fan like me, manual using is also possible. :)

**Comment by orenmizr** [comment]
i tried the accordion and the button as a test with shadcn-vue. the test was successful. i won';t know for sure until i will start using it. 

when shadcn-vue inits it creates a component file and points to tailwind.config... does that matter ? can i delete it ? should i point it to unocss.config ?

**Comment by zouhangwithsweet** [comment]
I think you should remove all packages & config files which are related to `tailwindcss`

**Comment by sadeghbarati** [comment]
After testing UnoCSS I couldn't configure it to have same style and same behaviour as cva or tailwind-variant

https://stackblitz.com/edit/vitejs-vite-jstrsg 


https://github.com/unocss/unocss/issues/2748
https://github.com/unpreset/unocss-transformer-alias

---

found a Neat UnoCSS Project 

https://github.com/pondrax/floeui/blob/master/src/lib/ui/shortcuts/button.js

in this case, we **don't need** libs like **`cva`** or **`tailwind-variants`**


#### Combine UnoCSS `rules` and `shortcuts` (simple)
```ts
import { defineConfig } from 'unocss'
import presetUno from '@unocss/preset-uno'
import presetAttributify from '@unocss/preset-attributify'
import { fonts } from '@unocss/preset-mini/rules'

export default defineConfig({
	presets: [
	        presetUno(),
		presetAttributify(),
		{
			name: 'shadcn-unocss',
			rules: [
				[/^button-(.*)$/, fonts[1][1] as any]
			],
			shortcuts: [
				[
					'button',
					[
						'inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors data-[disabled]:pointer-events-none data-[disabled]:opacity-50 data-[focus-visible]:outline-none data-[focus-visible]:ring-2 data-[focus-visible]:ring-ring data-[focus-visible]:ring-offset-2'
					].join(' '),
					{ layer: 'components button' }
				],
			]
		}
	],
})
```

```vue
<script setup lang="ts">
import { Button } from '@/lib/registry/default/ui/button'
</script>

<template>
  <Button button="~ 5xl">Button</Button>
  <a href="#" button="~">A tag that act like a button</a>
  
  <!-- :x :)))) xdd -->
  <button button="~ primary sm">bruh sorry</button>
</template>
```

**Comment by orenmizr** [comment]
interesting link. i am still looking into the best way for using unocss+tailwind(wind preset) to create a shadcn theme config: 

.btn: (classA classB.....) 
.link: (classA classB.....)

**Comment by orenmizr** [comment]
> found a Neat UnoCSS Project
> 
> https://github.com/pondrax/floeui/blob/master/src/lib/ui/shortcuts/button.js
> 
> in this case, we **don't need** libs like **`cva`** or **`tailwind-variants`**
> 
> #### Combine UnoCSS `rules` and `shortcuts` (simple)
> ```ts
> import { defineConfig } from 'unocss'
> import presetUno from '@unocss/preset-uno'
> import presetAttributify from '@unocss/preset-attributify'
> import { fonts } from '@unocss/preset-mini/rules'
> 
> export default defineConfig({
> 	presets: [
> 	        presetUno(),
> 		presetAttributify(),
> 		{
> 			name: 'shadcn-unocss',
> 			rules: [
> 				[/^button-(.*)$/, fonts[1][1] as any]
> 			],
> 			shortcuts: [
> 				[
> 					'button',
> 					[
> 						'inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors data-[disabled]:pointer-events-none data-[disabled]:opacity-50 data-[focus-visible]:outline-none data-[focus-visible]:ring-2 data-[focus-visible]:ring-ring data-[focus-visible]:ring-offset-2'
> 					].join(' '),
> 					{ layer: 'components button' }
> 				],
> 			]
> 		}
> 	],
> })
> ```
> 
> ```
> <script setup lang="ts">
> import { Button } from '@/lib/registry/default/ui/button'
> </script>
> 
> <template>
>   <Button button="~ 5xl">Button</Button>
>   <a href="#" button="~">A tag that act like a button</a>
>   
>   
>   <button button="~ primary sm">bruh sorry</button>
> </template>
> ```

but you still need to go over all the components one by one and add them to the theme config. 
i thought that was shadcn preset was all about.  whenever they add css the preset gets updated.

**Comment by buditanrim** [comment]
Hello, have we solved this? I run into a situation where I'd want to use shadcn-vue but not quite sure how to configure it with Unocss

**Comment by Saeid-Za** [comment]
I've been using UnoCSS with shad-vue for a while now, I have not found any problem so far, is there a blocker that anyone know of? I'd be happy to help !

**Comment by sadeghbarati** [comment]
@Saeid-Za We are welcomed any help here 💚

**Comment by therealokoro** [comment]
Ladies and gentlemen!.....

drum rolls🥁🥁🥁

May I present to you......... [Unocss Preset Shadcn](https://github.com/hyoban/unocss-preset-shadcn).

Your questions have been answered, your savior is here 😁

**Comment by NamesMT** [comment]
UnoCSS is no longer usable on v0.11

Edit:
For more context, it was previously usable through some workaround, e.g:

https://github.com/NamesMT/starter-fullstack
https://github.com/hyoban-template/shadcn-vue-unocss-starter

**Comment by onmax** [comment]
Could we create an option in Nuxt Module to not force tailwindcss installation? Maybe something like: 

```ts
{
	// other options,
	
	// @default tailwindcss
	cssLibrary: "tailwindcss" | "unocss"
}
```

https://github.com/unovue/shadcn-vue/blob/dev/packages/module/src/module.ts#L63

Current workaround is to run `pnpm patch`.

**Comment by peterbud** [comment]
@onmax with the latest release it seems it's working with UnoCSS and does not try to install taliwindcss

**Comment by clddup** [comment]
@Saeid-Za   Is there a simplified project demo? Thank you very much.

**Comment by Saeid-Za** [comment]
@clddup Yes, I'll make a Repo and share a link.

**Comment by clddup** [comment]
> [@clddup](https://github.com/clddup) Yes, I'll make a Repo and share a link.

Thank you so much

**Comment by pgiblock** [comment]
For everyone that has been using this with UnoCSS (And I presume unocss-preset-shadcn) how did you even get past the install? When I run `pnpm dlx shadcn-vue@latest init` I get:
```
No Tailwind CSS configuration found at /saturn.                                                                                                                                                                
It is likely you do not have Tailwind CSS installed or have an invalid configuration.                                                                                                                          
Install Tailwind CSS then try again.                                                                                                                                                                           
Visit https://tailwindcss.com/docs/guides/vite to get started.
```

I created an empty tailwind.config.js and/or ts file in my project root. I tried loading it with a base tailwind config such as:
```js
module.exports = {
  purge: [],
  darkMode: false,
  theme: {
    extend: {},
  },
  variants: {
    extend: {},
  },
  plugins: [],
}
```

But it seems I must install tailwind even if I won't be using it?

EDIT:
I reread the unocss-preset-shadcn instructions and realized they say not to `init`. That's all and well, but it seems the same check is performed with the `add` subcommand:

```
pnpm dlx shadcn-vue@latest add button
```
Leads to the same error.
